core/client/fs/src/main/java/alluxio/client/file/MetadataCache.java

```diff
@@ -100,15 +99,25 @@ public URIStatus get(AlluxioURI path) {
    * @param status the status to be cached
    */
   public void put(AlluxioURI path, URIStatus status) {
-    mCache.put(path.getPath(), new CachedItem(status));
+    put(path.getPath(), status);
   }
 
   /**
    * @param path the Alluxio path
    * @param status the status to be cached
    */
   public void put(String path, URIStatus status) {
-    mCache.put(path, new CachedItem(status));
+    try {
+      CachedItem item = mCache.get(path, new Callable<CachedItem>() {
+        @Override
+        public CachedItem call() throws Exception {
+          return new CachedItem();
+        }
+      });
+      item.setStatus(status);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
```

Comment:
same comments about logging and catching exceptions here.

Reply:
Sure. It should be ExecutionException and log the error info.